R/pkg-knitr.R

```diff
@@ -0,0 +1,157 @@
+#' knit print polars DataFrame
+#'
+#' Mimics [python-polars' NotebookFormatter]
+#' (https://github.com/pola-rs/polars/blob/4ffcb7461302a580255a9d910d70f1f7b2508675/py-polars/polars/dataframe/_html.py)
+#' for HTML outputs.
+#' @name knit_print.DataFrame
+#' @param x a polars DataFrame to knit_print
+#' @param ... additional arguments, not used
+#' @keywords DataFrame
+#' @export
+knit_print.DataFrame <- function(x, ...) {
+  .print_opt = getOption("polars.df_print", "auto")
+  if (.print_opt != "default" && (knitr::is_html_output() || (.print_opt == "html"))) {
```

Comment:
do we need to check if `knitr` is installed?

Reply:
As I understand it, this is a function of knitr and it is never used if knitr is not installed, so no check is needed.
I believe nanoarrow does the same thing already.

https://github.com/pola-rs/r-polars/blob/71e4e58d645ca5dcc11a9916803c70f90e1a481e/R/zzz.R#L135-L139